extension/solarwindsapmsettingsextension/extension.go

```diff
@@ -25,13 +32,63 @@ func newSolarwindsApmSettingsExtension(extensionCfg *Config, logger *zap.Logger)
 	return settingsExtension, nil
 }
 
-func (extension *solarwindsapmSettingsExtension) Start(_ context.Context, _ component.Host) error {
+func (extension *solarwindsapmSettingsExtension) Start(ctx context.Context, _ component.Host) error {
 	extension.logger.Debug("Starting up solarwinds apm settings extension")
-	_, extension.cancel = context.WithCancel(context.Background())
+	ctx, extension.cancel = context.WithCancel(ctx)
+	configOk := validateSolarwindsApmSettingsExtensionConfiguration(extension.config, extension.logger)
+	if configOk {
+		extension.conn, _ = grpc.Dial(extension.config.Endpoint, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
+		extension.logger.Info("Dailed to " + extension.config.Endpoint)
+		extension.client = collectorpb.NewTraceCollectorClient(extension.conn)
+		go func() {
+			ticker := newTicker(extension.config.Interval)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					refresh(extension)
+				case <-ctx.Done():
+					extension.logger.Debug("Received ctx.Done() from ticker")
+					return
+				}
+			}
+		}()
+	} else {
+		extension.logger.Warn("solarwindsapmsettingsextension is in noop. Please check config")
+	}
 	return nil
 }
 
 func (extension *solarwindsapmSettingsExtension) Shutdown(_ context.Context) error {
 	extension.logger.Debug("Shutting down solarwinds apm settings extension")
+	if extension.conn != nil {
+		return extension.conn.Close()
```

Comment:
you're not closing the ticker loop

Reply:
Followed the [tailtracerReceiver example](https://opentelemetry.io/docs/collector/building/receiver/#implementing-the-receiver-component) to add a call to the `cancel()`. 

Does it address the concern?

Reply:
that works, but you're still exiting early on this line without calling cancel.

Reply:
Thanks both! :)
I fixed the code.